app/models/check.rb

```diff
@@ -7,14 +7,12 @@ class Check < ApplicationRecord
           ]
 
   def state_machine
-    Statesman::Machine.retry_conflicts do
-      CheckStateMachine.new(
-        self,
-        transition_class: CheckTransition,
-        association_name: :check_transitions,
-        initial_transition: true
-      )
-    end
+    @state_machine ||= CheckStateMachine.new(
+      self,
+      transition_class: CheckTransition,
+      association_name: :check_transitions,
+      initial_transition: true
+    )
```

Comment:
Comme signalé par @Josyann, on a encore parfois des race conditions 🏇 qui sont un peu inévitables vu que l'état du contrôle est interrogé par le web et les workers en parallèle.

```suggestion
    Statesman::Machine.retry_conflicts do
      @state_machine ||= CheckStateMachine.new(
        self,
        transition_class: CheckTransition,
        association_name: :check_transitions,
        initial_transition: true
      )
    end
```